src/store/reducers/auth/autoLoginReducer.js

```diff
@@ -0,0 +1,26 @@
+// Actions
+import { FETCH_USER } from "../../actions/auth/autoLoginAction";
+
+const initialState = {
+  userData: {
+    authenticatedMethod: "",
+    userAvatar: "",
+    userEmail: "",
+    userId: undefined,
+    userName: "",
+    userType: "",
+  },
+};
```

Comment:
As the userData state in **authReducer** has already been in usage, adding another duplicated one will make it harder to avoid side effects. 
After the user logins, the **userData** is **already available.** So I **don't see a proper reason to fetch that data again from the database**
Suggestion: Remove this reducer to the already-created file: **authReducer**. @Chillisaucery

Reply:
the userData from your authReducer has 2 additional fields (uid and token), which made it not match the data on Firebase. If I just replace your userData with the userData from Firebase, your userData will not have those 2 fields after I fetch the data. As I don't know if you need those fields or not (right now in the code or in the future), I decided not to touch it.
![image](https://user-images.githubusercontent.com/66952546/94840443-a35bff00-0442-11eb-8e61-9f74cf55e076.png)

Reply:
Ngon